From 51a7dc6b47ed198eab69a5bb00deaf7a6ef3a30b Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 2 Aug 2026 13:38:03 +0200
Subject: [PATCH] ARM: dts: allwinner: sun8i-a83t-tbs-a711: Add pin bank
 supplies

None of the pin banks had a supply, so pinctrl falls back to a dummy
regulator for every one of them:

  sun8i-a83t-pinctrl 1c20800.pinctrl: supply vcc-pd not found, using
  dummy regulator
  sun8i-a83t-pinctrl 1c20800.pinctrl: supply vcc-pb not found, using
  dummy regulator
  [...]

The SoC has a dedicated VCC-PD input, the remaining banks are powered
from the common VCC-IO input. Both are connected to DCDC1 here.

PL is deliberately left out. r_rsb muxes PL0/PL1, so a bank supply on
the AXP813 would make the r_pio request defer on a regulator that only
shows up once the PMIC sitting behind that very bus has probed.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 .../arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
index 099377539ac6..1ae04c1219d3 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
@@ -445,6 +445,20 @@ &mmc2 {
 	status = "okay";
 };
 
+/*
+ * The SoC has a dedicated VCC-PD input, all the other banks are powered
+ * from the common VCC-IO input. Both are connected to DCDC1 on this board.
+ */
+&pio {
+	vcc-pb-supply = <&reg_dcdc1>;
+	vcc-pc-supply = <&reg_dcdc1>;
+	vcc-pd-supply = <&reg_dcdc1>;
+	vcc-pe-supply = <&reg_dcdc1>;
+	vcc-pf-supply = <&reg_dcdc1>;
+	vcc-pg-supply = <&reg_dcdc1>;
+	vcc-ph-supply = <&reg_dcdc1>;
+};
+
 &pwm {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pwm_pin>;
-- 
2.43.0

